packages/material-ui/src/StepLabel/StepLabel.js

```diff
@@ -96,13 +97,13 @@ function StepLabel(props) {
       )}
       {...other}
     >
-      {icon && (
+      {(icon || StepIconComponent) && (
         <span
```

Comment:
Isn't this now always true since we have a default value? I think this line does not need a change.